packages/discovery-provider/src/api/v1/users.py

```diff
@@ -1998,14 +1998,28 @@ def get(self):
             ns.abort(400, "JWT payload could not be decoded.")
 
         wallet_user_id = get_user_with_wallet(wallet)
-        if not wallet_user_id or (
-            wallet_user_id != payload["userId"]
-            and wallet_user_id != decode_string_id(payload["userId"])
-        ):
-            ns.abort(
-                404,
-                "The JWT signature is invalid - the wallet does not match the user.",
+        jwt_user_id = decode_string_id(payload["userId"])
+
+        if not wallet_user_id:
+            ns.abort(404, "The JWT signature is invalid - invalid wallet")
+
+        # If the user id found in the token does not match the signer of the token,
+        # check if the signing user is a manager of that user. Otherwise, reject.
+        if wallet_user_id != jwt_user_id:
+            is_managed_user = False
+            managed_users = get_user_managers_with_grants(
+                GetUserManagersArgs(user_id=jwt_user_id, is_approved=True)
             )
+            for m in managed_users:
+                if m["manager"]["user_id"] == wallet_user_id:
+                    is_managed_user = True
+
+            # Not authorized
+            if not is_managed_user:
+                ns.abort(
+                    404,
```

Comment:
Nit, 404 is Not Found, just making sure that's actually what we want here?

Reply:
Yeah we want 403 for this and the other i didn't touch too probably.